### PR TITLE
feat: authentication migrated to redux

### PR DESCRIPTION
**Pull request by vrx29**
Live Preview : https://gimesto-ecom-git-redux-auth-vrx29.vercel.app/

Authentication code moved from context to redux

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,23 +10,27 @@ import {
...
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <ProductProvider>
-      <AuthProvider>
+    <Provider store={store}>
+      <ProductProvider>
+        {/* <AuthProvider> */}
```
Comment (nayyyhaa) [inline src/index.js]: can remove the commented code & remove auth provider from the imports.

**Review by nayyyhaa — APPROVED** [review]
Good work in migrating context code to redux code. Really like that you are using cookieHandler in your code. Nice Job! 👍

File: src/pages/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
+    } else if (!validateEmail(email)) {
+      setError("Please enter email in correct format");
+    } else if (password !== confirmPassword) {
+      setError("Confirm Password must be same as Password");
+    } else {
+      dispatch(
+        signUp({ firstName, lastName, email, password, confirmPassword })
+      );
+    }
+    // navigate(from, { replace: true }); //After success send back to previous router from where it was routed here
```
Comment (Developers-codz) [inline src/pages/SignUp/SignUp.jsx]: remove commented code if not needed

**Review by Developers-codz — APPROVED** [review]
Good work on migrating context to redux.
Approving for merge

**Pull request merged**